subprojects/gst-plugins-bad/sys/msdk/gstmsdkallocator_libva.c

```diff
@@ -704,6 +719,104 @@ gst_msdk_export_dmabuf_to_vasurface (GstMsdkContext * context,
   }
 }
 
+static VASurfaceID
+_get_va_surface (GstBuffer * buf, GstVideoInfo * info,
+    GstMsdkContext * msdk_context)
+{
+  VASurfaceID va_surface = VA_INVALID_ID;
+
+  if (info == NULL) {
+    va_surface = gst_va_buffer_get_surface (buf);
+  } else {
+    GstVideoMeta *vmeta;
+    GstMemory *mem;
+    guint i, fd;
+
+    mem = gst_buffer_peek_memory (buf, 0);
+    fd = gst_dmabuf_memory_get_fd (mem);
+    if (fd < 0)
+      return va_surface;
+
+    vmeta = gst_buffer_get_video_meta (buf);
+    if (vmeta) {
+      if (GST_VIDEO_INFO_FORMAT (info) != vmeta->format ||
+          GST_VIDEO_INFO_WIDTH (info) != vmeta->width ||
+          GST_VIDEO_INFO_HEIGHT (info) != vmeta->height ||
+          GST_VIDEO_INFO_N_PLANES (info) != vmeta->n_planes) {
+        GST_ERROR ("VideoMeta attached to buffer is not matching"
+            "the negotiated width/height/format");
+        return va_surface;
+      }
+      for (i = 0; i < GST_VIDEO_INFO_N_PLANES (info); ++i) {
+        GST_VIDEO_INFO_PLANE_OFFSET (info, i) = vmeta->offset[i];
+        GST_VIDEO_INFO_PLANE_STRIDE (info, i) = vmeta->stride[i];
+      }
+      GST_VIDEO_INFO_SIZE (info) = gst_buffer_get_size (buf);
+    }
+    /* export dmabuf to vasurface */
+    if (!gst_msdk_export_dmabuf_to_vasurface (msdk_context, info, fd,
+            &va_surface)) {
+      return VA_INVALID_ID;
+    }
+  }
+
+  return va_surface;
+}
+
+GstMsdkSurface *
+import_to_msdk_surface (GstBuffer * buf, GstMsdkContext * msdk_context,
+    GstVideoInfo * info)
```

Comment:
Please add "const" to this info parameter.  You will modify it inside this function. But according to your usage, you pass the input_state->in_info to this function, which is an important global state for encoder and should not be changed at will.

Reply:
I have added const to protect the input GstVideoInfo from being modified. Annother copied one is used to fill surface info. Thanks for correcting this unsafe usage.